typescript/sdk/src/warp/WarpCore.ts

```diff
@@ -643,13 +647,54 @@ export class WarpCore {
   /**
    * Ensure token amount is valid
    */
-  protected validateAmount(
+  protected async validateAmount(
     originTokenAmount: TokenAmount,
-  ): Record<string, string> | null {
+    destination: ChainNameOrId,
+    recipient: Address,
+  ): Promise<Record<string, string> | null> {
     if (!originTokenAmount.amount || originTokenAmount.amount < 0n) {
       const isNft = originTokenAmount.token.isNft();
       return { amount: isNft ? 'Invalid Token Id' : 'Invalid amount' };
     }
+
+    // Check the transfer amount is sufficient on the destination side
+
+    const originToken = originTokenAmount.token;
+
+    const destinationName = this.multiProvider.getChainName(destination);
+    const destinationToken =
+      originToken.getConnectionForChain(destinationName)?.token;
+    assert(destinationToken, `No connection found for ${destinationName}`);
+    const destinationAdapter = destinationToken.getAdapter(this.multiProvider);
+
+    // Convert the originTokenAmount to a destination amount
+    const destinationTokenAmount = destinationToken.amount(
+      convertDecimals(
```

Comment:
It's nbd but If you convert `minDestinationTransferAmount` to originToken's decimals and _before_ comparing it the origin amount, I think you can avoid the need for 2 `convertDecimals` calls. They're not expensive or anything but it's kinda verbose.

Reply:
clever, fixed!